### PR TITLE
[SPARK-7084] improve saveAsTable documentation

### PR DESCRIPTION
**Pull request by phatak-dev**

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -1085,6 +1085,9 @@ class DataFrame private[sql](
    * there is no notion of a persisted catalog in a standard SQL context.  Instead you can write
    * an RDD out to a parquet file, and then register that file as a table.  This "table" can then
    * be the target of an `insertInto`.
+   *
+   * Also note that this doesn't create a hive table, but instead creates a Spark data source table.
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: Maybe say

"Also note that while this function can persist the table metadata into Hive's metastore, the table will NOT be accessible from Hive."

**Comment by phatak-dev** [comment]
Updated.

**Comment by rxin** [comment]
Jenkins, test this please.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 2]
```diff
@@ -1085,6 +1085,9 @@ class DataFrame private[sql](
    * there is no notion of a persisted catalog in a standard SQL context.  Instead you can write
    * an RDD out to a parquet file, and then register that file as a table.  This "table" can then
    * be the target of an `insertInto`.
+   *
+   * Also note that while this function can persist the table metadata into Hive's metastore,
+   * the table will NOT be accessible from Hive.
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: There are multiple saveAsTable functions. Do you mind updating them as well?

**Comment by phatak-dev** [comment]
Added for other methods also.

**Comment by rxin** [comment]
Jenkins, test this please.

**Comment by rxin** [comment]
LGTM. I will merge it after Jenkins comes back happy.

**Comment by rxin** [comment]
FYI I submitted a small patch on top of this to add a link to a jira ticket: https://github.com/apache/spark/pull/6067

**Comment by chiyingyunhua** [comment]
请问，我使用spark1.3.1的sparksql中dataframe的一个saveastable方法存了一个dataframe，但是用hive读的时候出错了Failed with exception java.io.IOException:java.io.IOException: hdfs://namenode71:8020/user/hive/warehouse/zz5/part-00000 not a SequenceFile。这个怎么破？1.2是可以的

**Pull request closed without merging**